### ISSUE TITLE
Need to have error type such as native GraphQLError

### ISSUE THREAD

**Issue by mcun0s** [ExpediaGroup/graphql-kotlin#838]
**Is your feature request related to a problem? Please describe.**
I was trying to customize the error message but I can't because ExecutionResult.toGraphQLResponse function of com.expediagroup.graphql.spring.extensions.ResponseExtensionsKt.class has static type of response. 
`
fun ExecutionResult.toGraphQLResponse(): GraphQLResponse<*> {
    val data: Any? = getData<Any?>()
    val filteredErrors = if (errors?.isNotEmpty() == true) errors?.map { it.toGraphQLKotlinType() } else null
    val filteredExtensions = if (extensions?.isNotEmpty() == true) extensions else null
    return GraphQLResponse(data, filteredErrors, filteredExtensions)
}
`
I was trying to have errorType on response message, but it was not working due to above code.

**Describe the solution you'd like**
Could you consider to have same GraphQLError type with native GraphQLError type(in graphql package)?

I'm not sure why you have separated Kotlin error type, but it makes reduce data interaction from native GraphQLError.
I totally understand I can add additional data set into extension property. However I think it would be better if we can errorType have on main data set. If 'errorType' is under extension, it looks like sub data set, but I believe we can define that data main data level.

**Comment by dariuszkuc** [comment]
Hello! By "native" I assume you mean `graphql-java` `GraphQLError`? Our [GraphQLError](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/graphql-kotlin-types/src/main/kotlin/com/expediagroup/graphql/types/GraphQLError.kt) data class was modeled after the [GraphQL spec](http://spec.graphql.org/June2018/#sec-Errors) which describes that errors should only contain `message`, optional list of `locations`, optional list of `path` segments and optional `extensions` that can contain anything. We created it so we don't have to rely on the internals of `graphql-java` `toSpecification()` implementation and instead just use Jackson to correctly serialize the objects that are spec compliant.

From the spec:
> GraphQL services should not provide any additional entries to the error format since they could conflict with additional entries that may be added in future versions of this specification.
>
>NOTE:
>Previous versions of this spec did not describe the extensions entry for error formatting. While non‐specified entries are not violations, they are still discouraged.

related: https://github.com/graphql/graphql-spec/issues/708

--- 
side note: looks like we got outdated javadoc on the error as it does not implement graphql-java interface
